nucypher/utilities/versioning.py

```diff
@@ -0,0 +1,160 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+from abc import abstractmethod, ABC
+from typing import Dict, Tuple, Callable, NamedTuple
+
+
+class Versioned(ABC):
+    """Base class for serializable entities"""
+
+    _PARTS = 2  # bytes
+    _PART_SIZE = 2
+    _BRAND_SIZE = 2
+    _VERSION_SIZE = _PART_SIZE * _PARTS
+    _HEADER_SIZE = _BRAND_SIZE + _VERSION_SIZE
+    _MAXIMUM_VERSION = 99  # length
```

Comment:
Doesn't seem to be used

Reply:
Whoops, I did not mean to commit this change. Thanks for noticing.

Reply:
✔️